This relies on link.h, which is present in musl as well as glibc.

--- a/src/base/elf_mem_image.h
+++ b/src/base/elf_mem_image.h
@@ -43,7 +43,7 @@
 
 // Maybe one day we can rewrite this file not to require the elf
 // symbol extensions in glibc, but for right now we need them.
-#if defined(__ELF__) && defined(__GLIBC__) && !defined(__native_client__)
+#if defined(__ELF__) && !defined(__native_client__)
 
 #define HAVE_ELF_MEM_IMAGE 1
 
